### ISSUE TITLE
Specialization should allow for delegation to default trait method implementation

### ISSUE THREAD

**Issue by jazzfool** [rust-lang/rust#68309]
Given the following trait:
```rust
trait Trait {
    fn test(&self) { println!("default implementation"); }
}
```
To provide a default specialization on `Trait::test`, `test` must be redefined;
```rust
impl<T> Trait for T {
    default fn test(&self) { println!("default implementation"); }
}
```
However this violates the DRY principle. The solution here would be to provide a way to delegate the implementation of `test` to the already existing default implementation found in the trait definition.

Proposed syntax to allow for this delegation:
```rust
impl<T> Trait for T {
    default fn test(&self);
}
```

**Comment by jazzfool** [comment]
cc @Centril

**Comment by Centril** [comment]
I believe this is intentional fallout from https://github.com/rust-lang/rust/pull/64564 (cc @jonas-schievink).

Having some sort of forwarding syntax to recover DRY seems reasonable and `default fn test(&self);` isn't a bad one.

**Comment by jonas-schievink** [comment]
Yup, sounds reasonable to me. I noticed the amount of repetition when working on that PR too:

> One annoyance with doing this is that it sometimes requires users to copy-paste a provided trait method into an impl just to mark it default (ie. there is no syntax to forward this impl method to the provided trait method).

**Comment by Centril** [comment]
Syntactic support for `default fn test(&self);` in an `impl` should already be there after https://github.com/rust-lang/rust/pull/67131. To support it semantically we would need changes in lowering and beyond.
